src/LibHdb++Cassandra.cpp

```
@@ -1119,637 +1286,489 @@ void HdbPPCassandra::extract_and_bind_rw_values(CassStatement* statement,
-		{
-#ifdef _CASS_LIB_DEBUG
-			cout << __func__<< ": SUCCESS in query: " << query_str.str() << endl;
-#endif
-			const CassRow* row = cass_iterator_get_row(iterator);
-			const char * last_event_res;
-			size_t last_event_res_length;
-			cass_value_get_string(cass_row_get_column(row, 0), &last_event_res, &last_event_res_length);
-			last_event = string(last_event_res,last_event_res_length);
-			found = true;
-#ifdef _CASS_LIB_DEBUG
-			cout << __func__<< "(" << fqdn_attr_name << "): last event = " << last_event << endl;
-#endif
-    	}
-		cass_result_free(result);
-		cass_iterator_free(iterator);
+		DEBUG_MSG("SUCCESS in query: " << query_str.str() << endl);
+
+		const CassRow* row = cass_iterator_get_row(iterator);
+		const char * last_event_res;
+		size_t last_event_res_length;
+		cass_value_get_string(cass_row_get_column(row, 0), &last_event_res, &last_event_res_length);
+		last_event = string(last_event_res,last_event_res_length);
+
+		found = true;
+		DEBUG_MSG("(" << fqdn_attr_name << "): last event = " << last_event << endl);
 	}
 
+	cass_result_free(result);
+	cass_iterator_free(iterator);
 	cass_future_free(future);
 	cass_statement_free(statement);
-	if(!found)
-	{
-#ifdef _CASS_LIB_DEBUG
-		cout << __func__<< "(" << fqdn_attr_name << "): NO RESULT in query: " << query_str.str() << endl;
-#endif
-		return -1;
-	}
-	return 0;
+
+	if(!found) DEBUG_MSG("(" << fqdn_attr_name << "): NO RESULT in query: " << query_str.str() << endl);
+	TRACE_MSG("Leaving");
+	return found;
 }
 
-int HdbPPCassandra::insert_Attr(Tango::EventData *data, HdbEventDataType ev_data_type)
+void HdbPPCassandra::insert_Attr(Tango::EventData *data, HdbEventDataType ev_data_type)
 {
-#ifdef _CASS_LIB_DEBUG
-//	cout << __func__<< ": entering..." << endl;
-#endif
+	TRACE_MSG("Entering");
 
-    if(data == 0)
-    {
-        cerr << "HdbPPCassandra::insert_Attr(): data is a null pointer!" << endl;
-        return -1;
-    }
+	if(data == 0)
+	{
+		// TODO (SJ) assert this...
```

I would rather throw an exception...

---

Added. I assume this is allowed behaviour (null parameters).

---

By doing like that we allow it indeed.